src/serializer/functions.js

```diff
@@ -262,36 +279,41 @@ export class Functions {
       );
       invariant(additionalFunctionEffects);
       effects = additionalFunctionEffects.effects;
+      let oldWriteEffects = this.writeEffects.get(functionValue);
+      let removeIds = string => string.replace(/.*id:.*/g, "");
+      if (
+        oldWriteEffects &&
+        removeIds(oldWriteEffects.effects.toDisplayString()) !==
```

Comment:
A semantic call is made based on comparing display strings after applying a regex? That seems sketchy. (Also didn't see a test for it.) A safer and easier way out would be to disallow calling `__optimize` on the same function value twice for now.

Also, what if the two function values are different, but both point to the same underlying `$ECMAScriptCode`?
Not for this PR, but to revisit in the future: In what initial state should a nested optimized function be evaluated in? The state after the outer optimized function is done (appropriate for event handlers), or in the state right where the `__optimize` call happened (appropriate for functions passed to `.map` etc. on arrays), or in a way where the entire environment is made abstract (the default if we don't know better?)

Reply:
I forget, when can we get two function values pointing to the same underlying `ECMAScriptCode`? Is that when we have two functions that incidentally have the same source, or something to do with `bind` or some other case?

Reply:
For example, here. Nothing bad seems to happen, but is that guaranteed? https://prepack.io/repl.html#GYVwdgxgLglg9mABMAFASkQbwFCMQJwFMoR8lRJYF0sDjSkAmARkQGpEWBuRAX234A3AIb5krALzJ0XbCLHBGiKajSyA+urgAHWAFsYAL0IpgzNdk079Rk4otA